common-libs/internal-api-clients/src/services/RevocationApiService.ts

```diff
@@ -1,36 +1,40 @@
-import { profile } from '@affinidi/common'
+import { profile } from '@affinidi/tools-common'
 
+import { createClientMethods } from '../helpers/client'
+import { createDidAuthSession } from '../helpers/DidAuthManager'
+import { createClient, DidAuthConstructorOptions, GetParams, wrapWithDidAuth } from '../helpers/didAuthClientWrapper'
 import revocationSpec from '../spec/_revocation'
-import { ParseSpec } from '../types/openapiParser'
-import { BuildApiType } from '../types/typeBuilder'
-import DidAuthApiService, { DidAuthConstructorOptions } from './DidAuthApiService'
 
 type ConstructorOptions = DidAuthConstructorOptions & { revocationUrl: string }
 
-type ApiType = BuildApiType<ParseSpec<typeof revocationSpec>>
-
 type ReplaceFieldsWithAny<T> = {
   [key in keyof T]: any
 }
 
+const { CreateDidAuthRequest, ...otherMethods } = createClientMethods(revocationSpec)
+const clientMethods = wrapWithDidAuth(CreateDidAuthRequest, otherMethods)
```

Comment:
when we are wrapping the client methods with did auth; can we enable/disable did auth for specific endpoints?

for example revocation has an endpoint that can be used without did-auth not usd in SDK. But if we were to add that endpoint to this service class, how could we "not have did auth" for that specific endpoint?

with the old code you could either call;
```
this.execute
this.executeWithDidAuth
```

Reply:
```
const { CreateDidAuthRequest, MethodWithoutDidAuth1, MethodWithoutDidAuth2, ...methodsWithDidAuth } = createClientMethods(spec)
const wrappedMethodsWithAuth = wrapWithDidAuth(CreateDidAuthRequest, otherMethods)

class Service {
  private client
  private authenticatedClient

  constructor(...) {
    this.client = createClient({ MethodWithoutDidAuth1, MethodWithoutDidAuth2 }, ...)
    this.authenticatedClient = createClientWithAuth(wrappedMethodsWithAuth, didAuthSession, ...)
  }

  method1(...) {
    return this.client.MethodWithoutDidAuth1(...)
  }

  method3(...) {
    return this.authenticatedClient.MethodWithDidAuth3(...)
  }
}
```

Reply:
But the better option would probably to have separate service classes for authenticated and non-authenticated methods.

Which would also allow to call non-authenticated methods without having to supply a correct did / DidAuthAdapter (which would not be used anyway) to the service constructor.